### PR TITLE
User info class

### PR DESCRIPTION
**Pull request by SakiAkizuki**

### CODE REVIEW

File: StudyBuddy/app/src/main/java/uw/studybuddy/Events/EventCreation.java [hunk 1]
```diff
@@ -89,6 +89,31 @@ public void onClick(View v) {
...
+                String subject = subjectCreate.getText().toString().trim();
+/*
+        if(!TextUtils.isEmpty(course) && !TextUtils.isEmpty(description) && !TextUtils.isEmpty(location) && !TextUtils.isEmpty(subject)){
+            dataMap.put("course", course);
+            dataMap.put("description", description);
+            dataMap.put("location", location);
+            dataMap.put("subject", subject);
+
+
+            mDatabase.push().setValue(dataMap).addOnCompleteListener(new OnCompleteListener<Void>() {
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/Events/EventCreation.java]: We wanted to keep all database-related methods in the `FirebaseInstance` class. We already have a method in `FirebaseInstance` called `addNewEventToDatabase`that does this.

File: StudyBuddy/app/src/main/java/uw/studybuddy/MainActivity.java [hunk 2]
```diff
@@ -41,7 +56,13 @@
         NavigationView.OnNavigationItemSelectedListener {
 
     // Data required for the app
-    private UserInfo user;
+    private UserInfo user = new UserInfo();
+
+    DatabaseReference mUserRootRef = FirebaseDatabase.getInstance().getReference().child("Users")
```
Comment (Leksha) [inline StudyBuddy/app/src/main/java/uw/studybuddy/MainActivity.java:61]: Refactor to `FirebaseInstance` class.

File: StudyBuddy/app/src/main/res/layout/activity_register.xml [hunk 3]
```diff
@@ -40,7 +40,7 @@
         android:background="@color/white"
         android:ems="10"
         android:fontFamily="sans-serif-smallcaps"
-        android:hint="Confirmation email address  "
+        android:hint="Please Type in Your UW Ques ID"
```
Comment (Leksha) [inline StudyBuddy/app/src/main/res/layout/activity_register.xml]: - Quest 
- Just use the `EmptyEmail` string instead

**Review by Leksha — CHANGES_REQUESTED** [review]

File: StudyBuddy/app/src/main/java/uw/studybuddy/MainActivity.java (hunk 2 above)
Comment (SakiAkizuki) [reply StudyBuddy/app/src/main/java/uw/studybuddy/MainActivity.java:61]: the mUserRootRef have to set here for the value change listener function

Comment (Leksha) [reply StudyBuddy/app/src/main/java/uw/studybuddy/MainActivity.java:61]: I meant that this code `FirebaseDatabase.getInstance().getReference().child("Users")` should be in `FirebaseInstance` class

We don't want any database-related code like this part `.child("Users")`in other places.

**Pull request merged**